lexer.py

```diff
@@ -0,0 +1,127 @@
+from tokens import TokenTypes, Token
+
+###############################################################################
+#                                                                             #
+#   LEXER                                                                     #
+#                                                                             #
+###############################################################################
+
+ESCCHARS = {
+    "n":  "\n",
+    "t":  "\t",
+    "\\": "\\",
+    "'": "'",
+    "r":  "\r",
+}
+
+RESERVED_WORDS = {
+            "SELECT",
+            "FROM",
+            "EDGES",
+            "NODES",
+            "TRAVERSE",
+            "BY",
+            "DEPTH",
+            "BREADTH",
+            "WHERE",
+            "LIMIT",
+}
+
+class Lexer(object):
+    def __init__(self, text):
+        self.text = text         # string input
+        self.pos = 0             # self.pos is index into self.text
+        self.line = 1            # current line
+        self.current_char = self.text[0] # current char
+        self.tokens = []         # tokens so far
+    def _error(self, msg=""):
+        raise Exception("LexingError, line %s: " + msg + "." % self.line)
```

Comment:
Why not actually create a `LexingError` exception class instead of throwing a generic `Exception`?  Since if we'll ever need to catch this exception, it'll be nicer to catch a `LexingError` rather than any `Exception` and then checking the error message to see if it's a lexing error.

Reply:
Also, don't mix string concatenation and interpolation.  Use one or the other.